tests/client/test_client_tenant_token.py

```diff
@@ -0,0 +1,94 @@
+# pylint: disable=invalid-name
+
+from re import search
+import pytest
+import meilisearch
+from tests import BASE_URL, MASTER_KEY
+from meilisearch.errors import MeiliSearchApiError
+import datetime
+
+def test_generate_tenant_token_with_search_rules(get_private_key, index_with_documents):
+    """Tests create a tenant token with only search rules."""
+    index_with_documents()
+    key = get_private_key
+
+    client = meilisearch.Client(BASE_URL, key['key'])
+
+    token = client.generate_tenant_token(search_rules=["*"])
+
+    token_client = meilisearch.Client(BASE_URL, token)
+    response = token_client.index('indexUID').search('')
+    assert isinstance(response, dict)
+    assert len(response['hits']) == 20
+    assert response['query'] == ''
+
+def test_generate_tenant_token_with_api_key(client, get_private_key, index_with_documents):
+    """Tests create a tenant token with only search rules."""
+    index_with_documents()
+    key = get_private_key
+
+    token = client.generate_tenant_token(search_rules=["*"], api_key=key['key'])
+
+    token_client = meilisearch.Client(BASE_URL, token)
+    response = token_client.index('indexUID').search('')
+    assert isinstance(response, dict)
+    assert len(response['hits']) == 20
+    assert response['query'] == ''
+
+def test_generate_tenant_token_with_expires_at(client, get_private_key, index_with_documents):
+    """Tests create a tenant token with only search rules."""
+    index_with_documents()
+    key = get_private_key
+
+    client = meilisearch.Client(BASE_URL, key['key'])
+
+    tomorrow = datetime.datetime.now() + datetime.timedelta(days=1)
+    timestamp = datetime.datetime.timestamp(tomorrow)
+
+    token = client.generate_tenant_token(search_rules=["*"], expires_at=int(timestamp))
+
+    token_client = meilisearch.Client(BASE_URL, token)
+    response = token_client.index('indexUID').search('')
+    assert isinstance(response, dict)
+    assert len(response['hits']) == 20
+    assert response['query'] == ''
+
+def test_generate_tenant_token_without_search_rules(get_private_key, index_with_documents):
+    """Tests create a tenant token with only search rules."""
+    index_with_documents()
+    key = get_private_key
+
+    client = meilisearch.Client(BASE_URL, key['key'])
+
+    token = client.generate_tenant_token(search_rules='')
+
+    token_client = meilisearch.Client(BASE_URL, token)
+    with pytest.raises(MeiliSearchApiError):
+        token_client.index('indexUID').search('')
+
+def test_generate_tenant_token_with_master_key(client, get_private_key, index_with_documents):
```

Comment:
I mention here: https://github.com/meilisearch/meilisearch-ruby/pull/299/files#r812021121 about this use case. I think this is the responsibility of the Meilisearch server to validate this rule, not ours (SDKs) 🦺 !

Reply:
See this [comment](https://github.com/meilisearch/meilisearch-python/pull/412#discussion_r812969705).